bin/lang-js/BUCK

```diff
@@ -1,113 +1,84 @@
 load(
     "@prelude-si//:macros.bzl",
-    "eslint",
+    "alias",
     "export_file",
-    "ts_test",
     "nix_omnibus_pkg",
-    "node_pkg_bin",
-    "npm_bin",
-    "package_node_modules",
-    "prettier_check",
-    "typescript_check",
-    "typescript_dist",
-    "typescript_runnable_dist",
-    "typescript_runnable_dist_bin",
 )
 load(
-    "@prelude-si//:pnpm.bzl",
-    "pnpm_task_library",
-    "pnpm_task_binary",
-    "pnpm_task_test",
+    "@prelude-si//:deno.bzl",
+    "deno_compile",
+    "deno_format",
+    "deno_test",
 )
 
 export_file(
     name = "package.json",
 )
 
-package_node_modules(
-    name = "node_modules",
-)
-
-filegroup(
-    name = "src",
-    srcs = glob([
-        "src/**/*.ts",
-        "tsconfig.json",
-    ]),
-)
-
-filegroup(
-    name = "test_src",
-    srcs = glob([
-        "tests/**/*",
-    ]),
-)
-
-prod_deps_srcs = {
-    "lib/ts-lib": "//lib/ts-lib:src",
-}
-
-dev_deps_srcs = {
-    "lib/eslint-config": "//lib/eslint-config:src",
-    "lib/tsconfig": "//lib/tsconfig:src",
-}
-
-npm_bin(
-    name = "tsup",
-)
-
-typescript_dist(
-    name = "dist",
-    srcs = [":src"],
-    tsup = ":tsup",
-    prod_deps_srcs = prod_deps_srcs,
-    dev_deps_srcs = dev_deps_srcs,
+export_file(
+    name = "deno.json",
 )
 
-typescript_runnable_dist(
+alias(
     name = "lang-js",
+    actual = ":bin"
 )
```

Comment:
This works with the `alias` rule, but I believe you could also drop this `alias` rule and rename the `deno_compile` below to have `name = "lang-js",` and then the `bin/lang-js:lang-js target would be the `deno_compile` rule directly. If memory serves I had an `alias` rule in because of all the dist/runnable_dist/whatever stack of rules to make the prior version work.